fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestSetup.kt

```diff
@@ -0,0 +1,50 @@
+package org.mozilla.fenix.helpers
+
+import android.util.Log
+import kotlinx.coroutines.runBlocking
+import mozilla.appservices.places.BookmarkRoot
+import mozilla.components.browser.storage.sync.PlacesBookmarksStorage
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Before
+import org.mozilla.fenix.helpers.Constants.TAG
+import org.mozilla.fenix.helpers.TestHelper.appContext
+import org.mozilla.fenix.ui.robots.notificationShade
+
+open class TestSetup {
+    lateinit var mockWebServer: MockWebServer
+    private val bookmarksStorage = PlacesBookmarksStorage(appContext.applicationContext)
+
+    @Before
+    fun setUp() {
+        Log.i(TAG, "TestSetup: Starting the @Before setup")
+        // Clear pre-existing notifications
+        notificationShade {
+            cancelAllShownNotifications()
+        }
+        runBlocking {
+            // Reset locale to EN-US if needed.
+            AppAndSystemHelper.resetSystemLocaleToEnUS()
+            // Check and clear the downloads folder
+            AppAndSystemHelper.clearDownloadsFolder()
+            // Make sure the Wifi and Mobile Data connections are on
+            AppAndSystemHelper.setNetworkEnabled(true)
+            // Clear bookmarks left after a failed test
+            val bookmarks = bookmarksStorage.getTree(BookmarkRoot.Mobile.id)?.children
+            Log.i(TAG, "Before cleanup: Bookmarks storage contains: $bookmarks")
+            bookmarks?.forEach { bookmarksStorage.deleteNode(it.guid) }
```

Comment:
try using a `suspend coroutine` here, and let's see if this fixes it. it _should_ only continue after we get confirmation the `delete` instruction as been commited and applied to the data store. 🤞 

```
import kotlin.coroutines.resume
import kotlin.coroutines.resumeWithException
import kotlin.coroutines.suspendCoroutine

private suspend fun deleteBookmark(guid: String) {
        suspendCoroutine<Unit> { continuation ->
            bookmarksStorage.deleteNode(guid) { success ->
                if (success) {
                    continuation.resume(Unit)
                } else {
                    continuation.resumeWithException(Exception("Failed to delete bookmark with GUID $guid"))
                }
            }
        }
    }
```